src/dos/cdrom_ioctl_linux.cpp

```diff
@@ -83,26 +269,47 @@ bool CDROM_Interface_Ioctl::ReadSectors(PhysPt buffer,
 		if ((Bitu)ret != buflen)
 			ret = -1;
 	}
-	close(cdrom_fd);
 
 	MEM_BlockWrite(buffer, buf.data(), buflen);
 
 	return (ret > 0);
 }
 
-bool CDROM_Interface_Ioctl::SetDevice(const char *path, const int cd_number)
+bool CDROM_Interface_Ioctl::Open(const char* device_name)
 {
-	bool success = CDROM_Interface_SDL::SetDevice(path, cd_number);
+	int fd = open(device_name, O_RDONLY | O_NONBLOCK);
+	if (fd == -1) {
+		return false;
+	}
+	if (IsOpen()) {
+		close(cdrom_fd);
+	}
+	cdrom_fd = fd;
+	return true;
+}
+
+bool CDROM_Interface_Ioctl::SetDevice(const char* path, const int cd_number)
+{
+	assert(path != nullptr);
+
+	int num = SDL_CDNumDrives();
+	if ((cd_number >= 0) && (cd_number < num)) {
+		auto cd_name = SDL_CDName(cd_number);
+		if (cd_name && Open(cd_name)) {
+			InitAudio(cd_number);
+			return true;
+		}
+	}
 
-	if (success) {
-		const char *tmp = SDL_CDName(cd_number);
-		if (tmp)
-			safe_strncpy(device_name, tmp, 512);
-		else
-			success = false;
+	for (int i = 0; i < num; i++) {
+		auto cd_name = SDL_CDName(i);
+		if (cd_name && strcmp(cd_name, path) == 0 && Open(cd_name)) {
+			InitAudio(i);
+			return true;
+		}
 	}
 
```

Comment:
Maybe log a warning if we couldn't find it?

Reply:
warned